test/infra.go

```diff
@@ -191,6 +191,23 @@ func RunApp(t *testing.T, appName string, env ...string) (string, string) {
 	return stdoutText, stderrText
 }
 
+// RunAppWithoutTestMode runs the application without setting IN_OTEL_TEST=true
+// This allows the app to use real exporters (console, OTLP, etc.) instead of ManualReader
+func RunAppWithoutTestMode(t *testing.T, appName string, env ...string) (string, string) {
```

Comment:
Instead of adding a new `RunAppWithoutTestMode`, we can modify the existing `RunApp` function. It should default to setting IN_OTEL_TEST=true if the `env` parameter is not specified by the user. If the user explicitly sets it, their value should be used instead.